src/servers/src/http.rs

```diff
@@ -352,65 +356,106 @@ pub struct ApiState {
     pub script_handler: Option<ScriptHandlerRef>,
 }
 
-impl HttpServer {
-    pub fn new(
-        sql_handler: ServerSqlQueryHandlerRef,
-        grpc_handler: ServerGrpcQueryHandlerRef,
-        options: HttpOptions,
-    ) -> Self {
+#[derive(Default)]
+pub struct HttpServerBuilder {
+    inner: HttpServer,
+}
+
+impl HttpServerBuilder {
+    pub fn new(options: HttpOptions) -> Self {
         Self {
-            sql_handler,
-            grpc_handler,
-            options,
-            opentsdb_handler: None,
-            influxdb_handler: None,
-            prom_handler: None,
-            user_provider: None,
-            script_handler: None,
-            shutdown_tx: Mutex::new(None),
+            inner: HttpServer {
+                sql_handler: None,
+                grpc_handler: None,
+                options,
+                opentsdb_handler: None,
+                influxdb_handler: None,
+                prom_handler: None,
+                user_provider: None,
+                script_handler: None,
+                metrics_handler: None,
+                shutdown_tx: Mutex::new(None),
+            },
         }
     }
 
-    pub fn set_opentsdb_handler(&mut self, handler: OpentsdbProtocolHandlerRef) {
+    pub fn with_sql_handler(&mut self, handler: ServerSqlQueryHandlerRef) -> &mut Self {
+        debug_assert!(
```

Comment:
Well, if using builder pattern, we actually don't need all these `debug_assert!`s. The `debug_assert!` is added because originally all the handlers are inserted at runtime.